mathics/builtin/combinatorial.py

```diff
@@ -266,3 +266,923 @@ class RogersTanimotoDissimilarity(_BooleanDissimilarity):
+            'BinormalDistribution[rho]',
+            'UniformDistribution[{min, max}]',
+            'WeibullDistribution[alpha, beta, mu]',
+            'HyperbolicDistribution[lambda, alpha, beta, delta, mu]',
+            'BetaBinomialDistribution[alpha, beta, n]',
+            'BinormalDistribution[{mu1, mu2}, {sigma1, sigma2}, rho]',
+            'BirnbaumSaundersDistribution[alpha, lambda]',
+            'BetaDistribution[alpha, beta]',
+            'StudentTDistribution[mu, sigma, nu]',
+            'PERTDistribution[{min, max}, c, lambda]',
+            'DiscreteUniformDistribution[{imin, imax}]',
+            'ChiDistribution[nu]',
+            'PolyaAeppliDistribution[theta, p]',
+            'PascalDistribution[n, p]',
+            'ChiSquareDistribution[nu]',
+            'ExtremeValueDistribution[alpha, beta]',
+            'GumbelDistribution[alpha, beta]',
+            'ErlangDistribution[k, lambda]',
+            'WignerSemicircleDistribution[r]',
+            'InverseChiSquareDistribution[nu, zeta]',
+            'MeixnerDistribution[a, b, m, d]',
+            'LogisticDistribution[mu, beta]',
+            'TriangularDistribution[{min, max}]',
+            'PERTDistribution[{min, max}, c]',
+            'BenfordDistribution[b]',
+            'ExponentialPowerDistribution[k, mu, sigma]',
+            'LindleyDistribution[delta]',
+            'DagumDistribution[p, a, b]',
+            'ExpGammaDistribution[k, theta, mu]',
+            'TriangularDistribution[{min, max}, c]',
+            'CauchyDistribution[a, b]',
+            'BinormalDistribution[{sigma1, sigma2}, rho]',
+            'LogSeriesDistribution[theta]',
+            'ZipfDistribution[n, rho]',
+            'KDistribution[nu, w]',
+            'BetaPrimeDistribution[p, q, alpha, beta]',
+            'TukeyLambdaDistribution[lambda]',
+            'SinghMaddalaDistribution[q, a, b]',
+            'RayleighDistribution[sigma]',
+            'InverseChiSquareDistribution[nu]',
+            'WaringYuleDistribution[alpha, beta]',
+            'TukeyLambdaDistribution[lambda, mu, sigma]',
+            'GeometricDistribution[p]',
+            'NoncentralBetaDistribution[alpha, beta, delta]',
+            'NormalDistribution[mu, sigma]',
+            'WakebyDistribution[alpha, beta, gamma, delta, mu]']
+
+
+def Quantile(Builtin):
+    """
+    <dl>
+    <dt>'Quantile[$list$, $q$]'
+        <dd>gives the qth quantile of list'.      
+    </dl>
+
+    >> Quantile[{1, 2, 3, 4, 5, 6, 7}, 1/2]
+     = 4
+    """
+    
+    ##TODO: implement other patterns
+    
+    rules = {'Quantile[list_, q_]':'Sort[list, Less][[Ceiling[q*Length[list]]]]'}
+    
+
+
+class Distribution(Builtin):
+    ##todo
+        
+    def cdf_sym(self, symbol):
+
+
+
+    def expectation(self):
+
+
+
+    def probability(self, conditions):
+
+
+
+class DistributionParameterQ(Builtin):
+
+    def apply(self, expr, evaluation):
+        'DistributionParameterQ[expr]'
+    ##todo
+
```

Comment:
this is to check if the distribution is valid. First, it would check the distribution itself, eventually yielding argr message. Second, if the test passes, it will check numerical parameters for correctness. Symbolic params are assumed correct